File: app/src/main/java/com/example/liftoff/command/UnknownCommand.java
package com.example.liftoff.command;

/**
 * A default ${@code Command} implementation,
 * Used if the arguments provided to the application contain no
 * known commands.
 */
class UnknownCommand implements Command {

    /**
     * Nothing to check here
     * @return ${@code true}
     */
    @Override
    public boolean isValid() {
        return true;
    }

    /**
     * Simply prints an abort message
     */
    @Override
    public void execute() {
        System.out.println("Unknown command received. For usage help use the 'gradle run --args=\"help\"' command");
    }
}

File: src/main/scala/com/example/liftoff/command/UnknownCommand.scala
package com.example.liftoff.command

import zio.{Console, UIO, ZIO}

class UnknownCommand extends Command {

  override def isValid: UIO[Boolean] = ZIO.succeed(true)

  override def execute: UIO[Unit] = Console
    .printError("Unknown command received. For usage help use the 'sbt \"run help\"' command")
    .orElse(ZIO.succeed())
}
